version/version.go

```diff
@@ -11,3 +16,24 @@ var (
 	// the program at linking time.
 	Revision = ""
 )
+
+func init() {
+	if info, ok := debug.ReadBuildInfo(); ok {
```

Comment:
Could we check first `Revision` is empty to avoid extra call to build info?

Reply:
Done :tada:

Reply:
Does this information gets used when buildx is used as a module? Because in that case it would be showing version from "whatever project used it as a module"

Reply:
:scream: ah! I think you're right, ugh.

We could try and read through the modules first, and look for buildx in there, so we don't hit that? Frustratingly, `BuildInfo.Main.Version` doesn't get populated, so we can't pull from there.

Maybe we should drop this PR, and avoid clever guessing, since using buildx as a module is  definitely a valid use case (e.g. in compose, in docker desktop, etc).

Reply:
I think `vcs.*` is only populated at build time

Reply:
Yeah - but I think it's populated by whatever vcs information is in the build.

So, if compose uses buildx as a module, then with this patch, we'd populate `Revision` with compose's revision.

Reply:
Ah right yeah